src/Mvc/Mvc.ViewFeatures/src/ClientComponentMarker.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Components
+{
+    // **Component descriptor protocol**
+    // MVC serializes one or more components as comments in HTML.
+    // Each comment is in the form <!-- Blazor:<<Json>>--> for example { "type": "client", "typeName": "<<type>>", "assembly": "<<assembly>>", "parameterDefinitions": [{"name":"<<name>>", typeName="<<type>>", "assembly": "<<assembly>>"}], "parameterValues": [<<value>>] }
+    // Where <<Json>> has the following properties:
+    // 'type' indicates the marker type. It's value for client-side components is 'client'.
+    // 'typeName' the full type name for the rendered component.
+    // 'assembly' the assembly name for the rendered component.
+    // 'parameterDefinitions' an array that contains the definitions for the parameters including their names and types and assemblies.
+    // 'parameterValues' an array containing the parameter values.
+    // 'prerenderId' a unique identifier that uniquely identifies the marker to match start and end markers.
+    internal struct ClientComponentMarker
```

Comment:
How frequently are we going to hit the message size limit in SignalR - have we tested this?